src/feature_lib.cpp

```diff
@@ -0,0 +1,21 @@
+#include "feature_lib.h"
+#include "conversions.h"
+
+namespace featurelib {
+
+SiftDetector::SiftDetector() {}
+
+featurelib::DetectionResult SiftDetector::detect(const std::vector<std::uint8_t> &input_vector) {
+  cv::Mat image = vectorToMat(input_vector, height, width);
+  cv::cvtColor(image, image, cv::COLOR_RGB2GRAY);
+
+  std::vector<cv::KeyPoint> sift_key_points;
+  cv::Mat sift_descriptors;
+  cv::Mat mask(255 * cv::Mat::ones(height, width,CV_8U));
```

Comment:
You can put it in class field so it won't be recreated for each image. And don't forget to update its sizes in width and height setters.

Reply:
done